### PR TITLE
[spark] Fix GPUs per spark worker node calculation

### PR DESCRIPTION
**Pull request by WeichenXu123**
## Why are these changes needed?

Fix GPUs per spark worker node calculation:

In current master code, if `spark.task.resource.gpu.amount` is set to `0`, "GPUs per spark worker node" variable is set to `0` , but the value "GPUs per spark worker node"  should not be affected by `spark.task.resource.gpu.amount`. In many cases, users set `spark.task.resource.gpu.amount` to `0` to avoid spark SQL jobs using GPU, but the spark cluster worker nodes still have GPU devices.

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by WeichenXu123** [comment]
CC @jjyao

**Review by jjyao — APPROVED** [review]

**Comment by jjyao** [comment]
Test failure

File: python/ray/util/spark/utils.py [hunk 1]
```diff
@@ -357,10 +357,14 @@ def _get_num_physical_gpus():
...
-        raise RuntimeError(
+    except Exception:
+        _logger.warning(
             "Running command `nvidia-smi` for inferring GPU devices list failed."
-        ) from e
+        )
+        _logger.debug(
+            "'nvidia-smi --query-gpu=name --format=csv,noheader' command execution failed.",
+            exc_info=True,
+        )
```
Comment (WeichenXu123) [inline python/ray/util/spark/utils.py]: swallow exception:

For cpu-only cluster,`nvidia-smi` is not avaiable, return 0 in this case.

**Comment by WeichenXu123** [comment]
@jjyao done.

File: python/ray/util/spark/utils.py [hunk 2]
```diff
@@ -357,11 +357,17 @@ def _get_num_physical_gpus():
...
+        _logger.warning(
             "Running command `nvidia-smi` for inferring GPU devices list failed."
-        ) from e
-    return len(completed_proc.stdout.strip().split("\n"))
+        )
+        _logger.debug(
+            "'nvidia-smi --query-gpu=name --format=csv,noheader' command execution "
+            "failed.",
+            exc_info=True,
+        )
```
Comment (jjyao) [inline python/ray/util/spark/utils.py:369]: why both debug and warning? if this can happen and is expected, we can use info, otherwise warning or error.

**Pull request merged**